src/Razor/src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Microsoft.NET.Sdk.Razor.MvcApplicationPartsDiscovery.targets

```diff
@@ -0,0 +1,103 @@
+***********************************************************************************************
+-->
+
+<Project ToolsVersion="14.0">
+  <UsingTask
+    TaskName="Microsoft.AspNetCore.Razor.Tasks.DiscoverMvcApplicationParts"
+    AssemblyFile="$(RazorSdkBuildTasksAssembly)"
+    Condition="'$(RazorSdkBuildTasksAssembly)' != ''" />
+
+  <PropertyGroup>
+    <GenerateMvcApplicationPartsAttribute Condition="'$(GenerateMvcApplicationPartsAssemblyAttributes)' == '' AND '$(OutputType)' == 'Exe'">true</GenerateMvcApplicationPartsAttribute>
+
+    <CoreCompileDependsOn Condition="'$(GenerateMvcApplicationPartsAttribute)' == 'true' AND '$(DesignTimeBuild)' != 'true'">
+      _DiscoverMvcApplicationParts;
+      $(CoreCompileDependsOn);
+    </CoreCompileDependsOn>
+
+    <_MvcApplicationPartAttributeGeneratedFile>$(IntermediateOutputPath)$(TargetName).MvcApplicationPartsAssemblyInfo$(DefaultLanguageSourceExtension)</_MvcApplicationPartAttributeGeneratedFile>
+
+    <_DiscoverMvcApplicationPartsDependsOn>
+      ResolveAssemblyReferences;
+      _InitializeDiscoverMvcApplicationParts;
+    </_DiscoverMvcApplicationPartsDependsOn>
+  </PropertyGroup>
+
+  <Target Name="_InitializeDiscoverMvcApplicationParts">
+    <!-- Can be used by testassets which need to simulate a reference to Microsoft.AspNetCore.App -->
+    <PropertyGroup Condition="'$(_ReferencesMicrosoftAspNetCoreApp)' == ''">
+      <_ReferencesMicrosoftAspNetCoreApp>@(FrameworkReference->AnyHasMetadataValue('Identity', 'Microsoft.AspNetCore.App'))</_ReferencesMicrosoftAspNetCoreApp>
+    </PropertyGroup>
+  </Target>
+
+  <Target
+    Name="_DiscoverMvcApplicationParts"
+    Inputs="$(MSBuildAllProjects);$(ResolveAssemblyReferencesStateFile)"
+    Outputs="$(_MvcApplicationPartAttributeGeneratedFile)"
+    DependsOnTargets="$(_DiscoverMvcApplicationPartsDependsOn)"
+    Condition="'$(_ReferencesMicrosoftAspNetCoreApp)' == 'true'">
+
+    <ItemGroup>
+      <_MvcAssemblyName Include="Microsoft.AspNetCore.Mvc" />
+      <_MvcAssemblyName Include="Microsoft.AspNetCore.Mvc.Abstractions" />
+      <_MvcAssemblyName Include="Microsoft.AspNetCore.Mvc.ApiExplorer" />
+      <_MvcAssemblyName Include="Microsoft.AspNetCore.Mvc.Core" />
+      <_MvcAssemblyName Include="Microsoft.AspNetCore.Mvc.Cors" />
+      <_MvcAssemblyName Include="Microsoft.AspNetCore.Mvc.DataAnnotations" />
+      <_MvcAssemblyName Include="Microsoft.AspNetCore.Mvc.Formatters.Json" />
+      <_MvcAssemblyName Include="Microsoft.AspNetCore.Mvc.Formatters.Xml" />
+      <_MvcAssemblyName Include="Microsoft.AspNetCore.Mvc.Localization" />
+      <_MvcAssemblyName Include="Microsoft.AspNetCore.Mvc.NewtonsoftJson" />
+      <_MvcAssemblyName Include="Microsoft.AspNetCore.Mvc.Razor" />
+      <_MvcAssemblyName Include="Microsoft.AspNetCore.Mvc.RazorPages" />
+      <_MvcAssemblyName Include="Microsoft.AspNetCore.Mvc.TagHelpers" />
+      <_MvcAssemblyName Include="Microsoft.AspNetCore.Mvc.ViewFeatures" />
```

Comment:
Do we need to list everything here and maintain that? What would happen if *just* listed `Microsoft.AspNet.Mvc.Abstractions`?

What will we do if we do a big *squish* in the future? Make this depend on the TFM?

Reply:
>  What would happen if just listed Microsoft.AspNet.Mvc.Abstractions?

We want a way to avoid Mvc assemblies being treated as "ReferencesMvc" i.e. we don't want to treat `Microsoft.AspNetCore.Mvc.Core` as an application part.

> What will we do if we do a big squish in the future? Make this depend on the TFM?

I guess so. More importantly, I think it's generally ok assemblies listed here. If in 4.0, there's only Microsoft.AspNetCore.Mvc and none of the rest, the dependency lookup is still correct since `Microsoft.AspNetCore.Mvc.Core` would generally never appear in your dependency graph

Reply:
I assume then that this works correctly when a non-existent assembly is specified (it kinda has to).